No overload variant of "__add__" of "Timestamp" matches argument type "Series[Timedelta]"

```python
import pandas as pd

s = pd.Series(pd.to_timedelta([1, 2, 3], "h"))
pd.Timestamp(0) + s.cumsum()
```

with mypy

```
$ mypy tmp.py 
tmp.py:4:1: error: No overload variant of "__add__" of "Timestamp" matches argument type "Series[Timedelta]"  [operator]
tmp.py:4:1: note: Possible overload variants:
tmp.py:4:1: note:     def __add__(self, ndarray[tuple[int, ...], dtype[timedelta64[timedelta | int | None]]], /) -> ndarray[tuple[int, ...], dtype[datetime64[date | int | None]]]
tmp.py:4:1: note:     def __add__(self, timedelta | timedelta64[timedelta | int | None] | Tick, /) -> Timestamp
tmp.py:4:1: note:     def __add__(self, TimedeltaSeries, /) -> TimestampSeries
tmp.py:4:1: note:     def __add__(self, TimedeltaIndex, /) -> DatetimeIndex
Found 1 error in 1 file (checked 1 source file)
```

with pyright

```
$ pyright tmp.py 
/home/rscholz/github/randolf-scholz/tsdm/tmp.py
  /home/rscholz/github/randolf-scholz/tsdm/tmp.py:4:1 - error: Operator "+" not supported for types "Timestamp" and "Series[Timedelta]" (reportOperatorIssue)
  /home/rscholz/github/randolf-scholz/tsdm/tmp.py:4:1 - warning: Expression value is unused (reportUnusedExpression)
1 error, 1 warning, 0 informations 
```

```
mypy                          1.16.0
mypy_extensions               1.1.0
pandas                        2.3.0
pandas-stubs                  2.2.3.250527
pyright                       1.1.401
```

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -2334,6 +2334,13 @@ class TimedeltaSeries(Series[Timedelta]):
         **kwargs: Any,
     ) -> Timedelta: ...
     def diff(self, periods: int = ...) -> TimedeltaSeries: ...  # type: ignore[override] # pyright: ignore[reportIncompatibleMethodOverride]
+    def cumsum(
+        self,
+        axis: AxisIndex | None = ...,
+        skipna: _bool = ...,
+        *args: Any,
+        **kwargs: Any,
+    ) -> TimedeltaSeries: ...
 
 class PeriodSeries(Series[Period]):
     @property

--- a/tests/test_series.py
+++ b/tests/test_series.py
@@ -3813,3 +3813,14 @@ def test_series_items() -> None:
     s = pd.Series(data=[1, 2, 3, 4], index=["cow", "coal", "coalesce", ""])
     check(assert_type(next(s.items()), tuple[Hashable, int]), tuple)
     check(assert_type(s.items(), Iterator[tuple[Hashable, int]]), Iterator)
+
+
+def test_cumsum_timedelta() -> None:
+
+    s = pd.Series(pd.to_timedelta([1, 2, 3], "h"))
+    check(assert_type(s.cumsum(), "TimedeltaSeries"), pd.Series, pd.Timedelta)
+    check(
+        assert_type(pd.Timestamp(0) + s.cumsum(), "TimestampSeries"),
+        pd.Series,
+        pd.Timestamp,
+    )

